dbms_test.go

```diff
@@ -0,0 +1,163 @@
+package sqllexer
+
+import (
+	"encoding/json"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+type output struct {
+	Expected          string             `json:"expected"`
+	ObfuscatorConfig  *obfuscatorConfig  `json:"obfuscator_config,omitempty"`
+	NormalizerConfig  *normalizerConfig  `json:"normalizer_config,omitempty"`
+	StatementMetadata *StatementMetadata `json:"statement_metadata,omitempty"`
+}
+
+type outputList []*output
+
+func getSubdirectories(directory string) ([]string, error) {
+	var subdirs []string
+	entries, err := os.ReadDir(directory)
+	if err != nil {
+		return nil, err
+	}
+	for _, entry := range entries {
+		if entry.IsDir() {
+			subdirs = append(subdirs, entry.Name())
+		}
+	}
+	return subdirs, nil
+}
+
+// TestQueriesPerDBMS tests a preset of queries and expected output per DBMS
+// Test folder structure:
+// -- testdata
+//
+//	-- dbms_type
+//	  -- query_type
+//	    -- query_name.sql
+//	    -- query_name.expected
+func TestQueriesPerDBMS(t *testing.T) {
+	dbmsTypes := []DBMSType{
+		DBMSPostgres,
+	}
+
+	for _, dbms := range dbmsTypes {
+		// Get all subdirectories of the testdata folder
+		baseDir := filepath.Join("testdata", string(dbms))
+		// Get all subdirectories of the testdata folder
+		queryTypes, err := getSubdirectories(baseDir)
+		if err != nil {
+			t.Fatal(err)
+		}
+
+		for _, qt := range queryTypes {
+			dirPath := filepath.Join(baseDir, qt)
+			files, err := os.ReadDir(dirPath)
+			if err != nil {
+				t.Fatal(err)
+			}
+
+			for _, file := range files {
+				// the testdata folder contains only .sql and .expected files
+				// so we can safely ignore the other files
+				if strings.HasSuffix(file.Name(), ".sql") {
+					// Remove the .sql extension to get the test name
+					testName := strings.TrimSuffix(file.Name(), ".sql")
+					t.Run(testName, func(t *testing.T) {
```

Comment:
Nice polish of mapping the test name to the file name without the extension.

Reply:
that's actually the reason why I give the test files meaningful names instead of test1.sql or test2.sql 😆